### ISSUE TITLE
Windows 10, classic console, UTF-8, Consolas Font, Powershell 7

### ISSUE THREAD

**Issue by patrickbkr** [japhb/Terminal-Tests#7]
![grafik](https://user-images.githubusercontent.com/6307078/235861863-c0c9ccff-22ec-427b-8e04-9a59aedb9d69.png)
![grafik](https://user-images.githubusercontent.com/6307078/235862235-c987bc8a-90c5-4083-b8f5-05de023016be.png)
![grafik](https://user-images.githubusercontent.com/6307078/235862280-ad63cfab-53c9-484b-8e09-d30126cb198e.png)

**Comment by japhb** [comment]
Thank you for the screenshot!

Well that's odd.  Despite claiming UTF-8 support, it doesn't seem to be processing it properly -- but it isn't even handling ANSI attribute and color escape sequences properly, the first terminal I've seen with that limitation.  I'm guessing this is because of the "classic console"?

(Points for the choice of 8x16 as default character cell size though; Unicode has a number of glyphs that show up best in sizes divisible by 8.)

May I use these screenshots in the docs?

**Comment by patrickbkr** [comment]
Yes, use these screenshots to your liking.

Update: It does claim a UTF-8 codepage, but UTF-8 support wasn't enabled in Windows itself (as instructed [here](https://gist.github.com/sapeurfaire/91fe247cd9523971b8fab72d50df6c2a)).

**Comment by patrickbkr** [comment]
This screenshot is with the [default system locale set to use UTF-8](https://gist.github.com/sapeurfaire/91fe247cd9523971b8fab72d50df6c2a).
![grafik](https://user-images.githubusercontent.com/6307078/236788273-6b7716c7-547f-4aa4-9523-a0a45d00618c.png)

**Comment by japhb** [comment]
Still no ANSI escape sequence support, and seems to be limited to a glyph repertoire from around the WGL4 timeframe, but that's a huge improvement for UTF-8 handling.  :-)

**Comment by patrickbkr** [comment]
I'm a bit puzzled myself, since my conhost.exe should support most of this according to [this 2018 news post](https://devblogs.microsoft.com/commandline/new-experimental-console-features/).

**Comment by japhb** [comment]
Perhaps an option that you need to enable?

**Comment by patrickbkr** [comment]
> Perhaps an option that you need to enable?

I think I found the answer here: https://github.com/microsoft/WSL/issues/1173#issuecomment-254250445

A bigger read: [Classic Console APIs versus Virtual Terminal Sequences](https://learn.microsoft.com/en-us/windows/console/classic-vs-vt), especially [Exceptions for using Windows Console APIs](https://learn.microsoft.com/en-us/windows/console/classic-vs-vt#exceptions-for-using-windows-console-apis). (I find most of the Windows Console documentation to be a good read, especially the [Roadmap](https://learn.microsoft.com/en-us/windows/console/ecosystem-roadmap).)

In short each process that wants its output VT sequences to be processed needs to explicitly enable VT mode using [SetConsoleMode](https://learn.microsoft.com/en-us/windows/console/setconsolemode). I think (but don't know for sure) this only applies to conhost, but not Windows Terminal.

So in short: No I can't enable that, the Rakudo process itself needs to set that option. :-/
